pyigm/clustering/xcorr_utils.py

```diff
@@ -322,7 +386,175 @@ def spline_sensitivity(galreal, Nmin=20, magmin=17., magmax=26., delta_mag=0.5,
+            #Wr = logN_b_to_Wr(absreal.LOGN[i], absreal.B[i], ion='HI')
+            z = z_Lya
+            Wmin = Wmin_Lya
+
+        zgood = (Wr > Wmin) & (z >= zmin) & (z < zmax)
+
+        if np.sum(zgood) == 0:
+            from matplotlib import pyplot as plt
+            plt.plot(z, Wmin, drawstyle='steps-mid')
+            plt.axis([z[0], z[-1], 0, 0.1])
+            plt.show()
+            print(Wmin)
+        assert np.sum(zgood) > 0, \
+            'There are not regions in the spectrum with Wmin<{} A. The minimum is {}. Adjust significance.'.format(
+                Wr, np.min(Wmin))
+
+        # Random time
+        rand_z = RanDist(z, zgood * 1.)
+        zrand = rand_z.random(Nrand)
+        absrand.ZABS[i * Nrand:(i + 1) * Nrand] = zrand
+
+    return absrand
+
+def mask_galactic():
+    # masked regions (potential Galactic absorption)
+    galactic = np.array([1334.5323,  # CII
+                         1335.7077, # CII*
+                         1238.821,  # NV
+                         1242.804,  # NV
+                         1302.1685,  # OI
+                         #1304.8576,  # OI*
+                         #1306.0286,  # OI**
+                         1304.3702,  # SiII
+                         1260.4221,  # SiII
+                         1526.707,  # SiII
+                         1548.204,  # CIV
+                         1550.781,  # CIV
+                         1334.8132,  # PIII
+                         1259.519,  # SII
+                         1253.811,  # SII
+                         1250.584,  # SII
+                         1670.7886, # AlII
+                         1608.4511])  # FeII
+    return galactic
+
+
+def compute_Wmin(wa, fl, er, sl=3., R=20000, FWHM=10, ion='HI', dv_mask=200.):
+    """For a given spectrum and transition, it computes the minimun
+    rest-frame equivalent width for that transition to be observed. It
+    return a tuple of redshift and Wmin (z,Wmin)"""
+    from scipy.ndimage import uniform_filter as uf
+    #
+    if ion == 'HI':
+        w0 = 1215.67  # HI Lya w0 in angstroms
+    if ion == 'HILyb':
+        w0 = 1025.72  # HI Lyb w0 in angstroms
+    # Mask galactic
+    galactic = mask_galactic()
+    zgal = galactic / w0 - 1.
+    dzgal = (zgal + 1) * dv_mask / Ckms
+
+    wa = np.array(wa)
+    fl = np.array(fl)
+    er = np.array(er)
+
+    z = wa / w0 - 1.  # spectrum in z coordinates
+
+    Wmin = sl * w0 * er / R / fl  # sl*wa / (1. + z) / R / (S/N)
+    Wmin = np.where(Wmin <= 0, 1e10, Wmin)
+    Wmin = np.where(np.isnan(Wmin), 1e10, Wmin)
+    Wmin = np.where(np.isinf(Wmin), 1e10, Wmin)
+    Wmin = uf(Wmin.astype(float), FWHM)  # smoothed version (uniform preferred over gaussian)
+    for zi, dzi in zip(zgal, dzgal):
+        cond = (z > zi - dzi) & (z < zi + dzi)
+        Wmin = np.where(cond, 1e10, Wmin)
+    return z, Wmin
+
+def random_gal(galreal, Nrand, magbins, SPL, debug=False):
     """
     Preferred random galaxy generator.
 
```

Comment:
... sensitivity function estimated from galaxies with magnitudes within those defined by `magbins` edges (?)

Reply:
Add description of magbins.

Reply:
added